File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.c
int findLucky(int* arr, int arrSize) {
    int freq[501]={0};
    int index = 0;
    for(int i=0;i<arrSize;i++){
        freq[arr[i]]++;
        index = (arr[i]>index) ? arr[i] : index;
    }
    for(int i=index;i>=1;i--){
        if(freq[i]== i){
            return i;
        }
    }
    return -1;
}

File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.java
class Solution {
    public int findLucky(int[] arr) {
        Map<Integer,Integer> map = new HashMap<>();

        int lucky = -1;
        for(int i=0;i<arr.length;i++){
            map.put(arr[i],map.getOrDefault(arr[i],0)+1);
        }
        for(int i : map.keySet()){
            System.out.println(i);
            System.out.println(map.get(i));
            if(map.get(i) == i &&  i> lucky){
                lucky = i;
            }
        }
        return lucky;
    }
}